File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/inline/comment/direct/Java/Go/s504715611_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type UnionFind struct {
	Parent []int
}

func NewUnionFind(n int) *UnionFind {
	parent := make([]int, n)
	for i := range parent {
		parent[i] = -1
	}
	return &UnionFind{Parent: parent}
}

func (uf *UnionFind) Root(a int) int {
	if uf.Parent[a] < 0 {
		return a
	}
	uf.Parent[a] = uf.Root(uf.Parent[a])
	return uf.Parent[a]
}

func (uf *UnionFind) Size(a int) int {
	return -uf.Parent[uf.Root(a)]
}

func (uf *UnionFind) Connect(a, b int) bool {
	a = uf.Root(a)
	b = uf.Root(b)
	if a == b {
		return false
	}
	if uf.Size(a) < uf.Size(b) {
		a, b = b, a
	}
	uf.Parent[a] += uf.Parent[b]
	uf.Parent[b] = a
	return true
}

var (
	scanner = bufio.NewScanner(os.Stdin)
	reader  = strings.NewReader("")
)

func nextString() string {
	scanner.Scan()
	return scanner.Text()
}

func nextInt() int {
	scanner.Scan()
	i, _ := strconv.Atoi(scanner.Text())
	return i
}

func nextLong() int64 {
	scanner.Scan()
	i, _ := strconv.ParseInt(scanner.Text(), 10, 64)
	return i
}

func nextFloat() float64 {
	scanner.Scan()
	i, _ := strconv.ParseFloat(scanner.Text(), 64)
	return i
}

func main() {
	scanner.Split(bufio.ScanWords)
	str := nextString()
	if str[2] == str[3] && str[4] == str[5] {
		fmt.Println("Yes")
	} else {
		fmt.Println("No")
	}
}

func lowerBound(a []int, key int) int {
	right := len(a)
	left := 0
	for right-left > 1 {
		mid := (right + left) / 2
		if a[mid] < key {
			left = mid
		} else {
			right = mid
		}
	}
	return left
}

func upperBound(a []int, key int) int {
	right := len(a)
	left := 0
	for right-left > 1 {
		mid := (right + left) / 2
		if a[mid] <= key {
			left = mid
		} else {
			right = mid
		}
	}
	return left
}

func isPrime(n int64) bool {
	if n == 2 {
		return true
	}
	if n < 2 || n%2 == 0 {
		return false
	}
	d := int64(float64(n) ** 0.5)
	for i := int64(3); i <= d; i += 2 {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func upperDivision(a, b int) int {
	if a%b == 0 {
		return a / b
	}
	return a/b + 1
}

func lUpperDivision(a, b int64) int64 {
	if a%b == 0 {
		return a / b
	}
	return a/b + 1
}

func setArray(a int) []int {
	b := make([]int, a)
	for i := 0; i < a; i++ {
		b[i] = nextInt()
	}
	return b
}

func lSetArray(a int) []int64 {
	b := make([]int64, a)
	for i := 0; i < a; i++ {
		b[i] = nextLong()
	}
	return b
}

func reverse(s string) string {
	runes := []rune(s)
	for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
		runes[i], runes[j] = runes[j], runes[i]
	}
	return string(runes)
}

func printArray(que []int) {
	for i := 0; i < len(que)-1; i++ {
		fmt.Print(que[i], " ")
	}
	fmt.Println(que[len(que)-1])
}

func doubleSort(a [][]int) [][]int {
	return a
}

func lDoubleSort(a [][]int64) [][]int64 {
	return a
}

func modPow(x, n, mo int64) int64 {
	sum := int64(1)
	for n > 0 {
		if n&1 == 1 {
			sum = sum * x % mo
		}
		x = x * x % mo
		n >>= 1
	}
	return sum
}

func revCh(ch []rune) []rune {
	ret := make([]rune, len(ch))
	for i, j := len(ch)-1, 0; i >= 0; i, j = i-1, j+1 {
		ret[j] = ch[i]
	}
	return ret
}

func revInt(ch []int) []int {
	ret := make([]int, len(ch))
	for i, j := len(ch)-1, 0; i >= 0; i, j = i-1, j+1 {
		ret[j] = ch[i]
	}
	return ret
}

func warshallFloyd(v [][]int, n int) {
	for k := 0; k < n; k++ {
		for i := 0; i < n; i++ {
			for j := 0; j < n; j++ {
				v[i][j] = min(v[i][j], v[i][k]+v[k][j])
			}
		}
	}
}

func modInt(a, m int64) int64 {
	b := m
	u := int64(1)
	v := int64(0)
	for b != 0 {
		t := a / b
		a -= t * b
		x := a
		a = b
		b = x

		u -= t * v
		x = u
		u = v
		v = x
	}
	u %= m
	if u < 0 {
		u += m
	}
	return u
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func lMin(a, b int64) int64 {
	if a < b {
		return a
	}
	return b
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func lMax(a, b int64) int64 {
	if a > b {
		return a
	}
	return b
}

func gcd(a, b int64) int64 {
	if b > 0 {
		return gcd(b, a%b)
	}
	return a
}

func lcm(a, b int64) int64 {
	return a * b / gcd(a, b)
}

func modLcm(a, b, mod int64) int64 {
	return a * b * modInt(gcd(a, b), mod) % mod
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/C++/s504715611_commented.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

struct UnionFind {
    vector<int> Parent;
    UnionFind(int n) : Parent(n, -1) {}
    int root(int A) {
        if (Parent[A] < 0) return A;
        return Parent[A] = root(Parent[A]);
    }
    int size(int A) { return -Parent[root(A)]; }
    bool connect(int A, int B) {
        A = root(A);
        B = root(B);
        if (A == B) return false;
        if (size(A) < size(B)) swap(A, B);
        Parent[A] += Parent[B];
        Parent[B] = A;
        return true;
    }
};

struct FastScanner {
    char buffer[1024];
    int ptr, buflen;

    FastScanner() : ptr(0), buflen(0) {}

    bool hasNextByte() {
        if (ptr < buflen) return true;
        ptr = 0;
        buflen = fread(buffer, 1, 1024, stdin);
        return buflen > 0;
    }

    int readByte() {
        if (hasNextByte()) return buffer[ptr++];
        return -1;
    }

    bool hasNext() {
        skipWhite();
        return hasNextByte();
    }

    void skipWhite() {
        while (hasNextByte()) {
            int c = buffer[ptr];
            if (c == ' ' || c == '\n' || c == '\r' || c == '\t') {
                ptr++;
            } else {
                break;
            }
        }
    }

    string next() {
        skipWhite();
        if (!hasNextByte()) return "";
        string s;
        int b = buffer[ptr++];
        while (isPrintableChar(b)) {
            s += (char)b;
            if (hasNextByte()) {
                b = buffer[ptr++];
            } else {
                break;
            }
        }
        return s;
    }

    static bool isPrintableChar(int c) {
        return 33 <= c && c <= 126;
    }

    long long nextLong() {
        if (!hasNext()) return 0;
        long long n = 0;
        bool minus = false;
        int b = readByte();
        if (b == '-') {
            minus = true;
            b = readByte();
        }
        if (b < '0' || b > '9') return  bergen:: 0;
        while (true) {
            if ('0' <= b && b <= '9') {
                n = n * 10 + (b - '0');
            } else if (b == -1 || !isPrintableChar(b)) {
                break;
            } else {
                return 0;
            }
            b = readByte();
        }
        return minus ? -n : n;
    }

    int nextInt() { return (int)nextLong(); }

    double nextDouble() {
        string s = next();
        return stod(s);
    }

    char nextChar() {
        if (!hasNext()) return 0;
        int b = readByte();
        while (!isPrintableChar(b)) {
            b = readByte();
        }
        return (char)b;
    }
};

int lower_bound(int a[], int n, int key) {
    int left = 0, right = n;
    while (right - left > 1) {
        int mid = (left + right) / 2;
        if (a[mid] < key)
            left = mid;
        else
            right = mid;
    }
    return left;
}

int upper_bound(int a[], int n, int key) {
    int left = 0, right = n;
    while (right - left > 1) {
        int mid = (left + right) / 2;
        if (a[mid] <= key)
            left = mid;
        else
            right = mid;
    }
    return left;
}

bool isPrime(long n) {
    if (n == 2) return true;
    if (n < 2 || n % 2 == 0) return false;
    long d = sqrt(n);
    for (long i = 3; i <= d; i += 2)
        if (n % i == 0) return false;
    return true;
}

int upper_division(int a, int b) {
    return (a % b == 0) ? a / b : a / b + 1;
}

long lupper_division(long a, long b) {
    return (a % b == 0) ? a / b : a / b + 1;
}

vector<int> setArray(int a) {
    vector<int> b(a);
    for (int i = 0; i < a; i++)
        b[i] = scan.nextInt();
    return b;
}

vector<long> lsetArray(int a) {
    vector<long> b(a);
    for (int i = 0; i < a; i++)
        b[i] = scan.nextLong();
    return b;
}

string reverse(const string& str) {
    string s = str;
    int n = s.length();
    for (int i = 0; i < (n + 1) / 2; i++)
        swap(s[i], s[n - 1 - i]);
    return s;
}

void printArray(const vector<int>& que) {
    for (int i = 0; i < que.size() - 1; i++)
        cout << que[i] << " ";
    cout << que.back() << endl;
}

vector<vector<int>> doublesort(const vector<vector<int>>& a) {
    vector<vector<int>> sorted = a;
    sort(sorted.begin(), sorted.end(), [](const vector<int>& x, const vector<int>& y) {
        return x[0] < y[0];
    });
    return sorted;
}

vector<vector<long>> ldoublesort(const vector<vector<long>>& a) {
    vector<vector<long>> sorted = a;
    sort(sorted.begin(), sorted.end(), [](const vector<long>& x, const vector<long>& y) {
        return x[0] < y[0];
    });
    return sorted;
}

long long modpow(long long x, long long n, long long mo) {
    long long sum = 1;
    while (n > 0) {
        if (n & 1)
            sum = sum * x % mo;
        x = x * x % mo;
        n >>= 1;
    }
    return sum;
}

char* revch(char ch[], int len) {
    char* ret = new char[len];
    for (int i = 0, j = len - 1; i < len; i++, j--)
        ret[i] = ch[j];
    return ret;
}

int* revint(int ch[], int len) {
    int* ret = new int[len];
    for (int i = 0, j = len - 1; i < len; i++, j--)
        ret[i] = ch[j];
    return ret;
}

void warshall_floyd(int v[][100], int n) {
    for (int k = 0; k < n; k++)
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                v[i][j] = min(v[i][j], v[i][k] + v[k][j]);
}

long modint(long a, long m) {
    long b = m, u = 1, v = 0;
    while (b != 0) {
        long t = a / b;
        a -= t * b;
        long x = a;
        a = b;
        b = x;
        u -= t * v;
        x = u;
        u = v;
        v = x;
    }
    u %= m;
    if (u < 0) u += m;
    return u;
}

FastScanner scan;

int main() {
    string str = scan.next();
    if (str.length() < 6)
        cout << "No" << endl;
    else {
        if (str[2] == str[3] && str[4] == str[5])
            cout << "Yes" << endl;
        else
            cout << "No" << endl;
    }
    return 0;
} // End of Code
